explainability-connectors/src/test/java/org/kie/trustyai/connectors/kserve/v2/TensorConverterTest.java

```diff
@@ -0,0 +1,173 @@
+package org.kie.trustyai.connectors.kserve.v2;
+
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+import org.junit.jupiter.api.Test;
+import org.kie.trustyai.connectors.kserve.v2.grpc.InferTensorContents;
+import org.kie.trustyai.connectors.kserve.v2.grpc.ModelInferRequest;
+import org.kie.trustyai.connectors.kserve.v2.grpc.ModelInferResponse;
+import org.kie.trustyai.explainability.model.*;
+import org.kie.trustyai.explainability.model.TensorDataframe;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class TensorConverterTest {
+    @Test
+    void modelInferResponseToPredictionOutputSingle() {
+
+        final Random random = new Random();
+        final double value = random.nextDouble();
+
+        InferTensorContents.Builder contents = InferTensorContents.newBuilder()
+                .addFp64Contents(value);
+
+        ModelInferResponse.InferOutputTensor outputTensor = ModelInferResponse.InferOutputTensor.newBuilder()
+                .setDatatype("FP64")
+                .addShape(1).addShape(1).setContents(contents).build();
+
+        final ModelInferResponse response = ModelInferResponse.newBuilder().addOutputs(outputTensor).build();
+        PredictionOutput predictionOutput = TensorConverter.parseKserveModelInferResponse(response, 1).get(0);
+
+        assertEquals(1, predictionOutput.getOutputs().size());
+        assertEquals(value, predictionOutput.getOutputs().get(0).getValue().asNumber());
+    }
+
+    @Test
+    void modelInferResponseToPredictionOutputMulti() {
+
+        final Random random = new Random();
+        final List<Double> values = random.doubles(3).boxed().collect(Collectors.toList());
+        InferTensorContents.Builder contents = InferTensorContents.newBuilder()
+                .addFp64Contents(values.get(0))
+                .addFp64Contents(values.get(1))
+                .addFp64Contents(values.get(2));
+
+        ModelInferResponse.InferOutputTensor outputTensor = ModelInferResponse.InferOutputTensor.newBuilder()
+                .setDatatype("FP64")
```

Comment:
It should be nice to have the possible types in some enum or similar but I don't think we can easily customise these generated classes so let's stay with this for now. We can create a follow up ticket if we agree